src/invoices/Voucher/VoucherEditForm/VoucherEditContainer.js

```diff
@@ -0,0 +1,107 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import ReactRouterPropTypes from 'react-router-prop-types';
+import { get } from 'lodash';
+
+import {
+  Paneset,
+  LoadingView,
+} from '@folio/stripes/components';
+import {
+  stripesConnect,
+} from '@folio/stripes/core';
+import {
+  getConfigSetting,
+  useShowCallout,
+} from '@folio/stripes-acq-components';
+
+import {
+  invoiceResource,
+  VOUCHER_BY_ID,
+  configVoucherNumber,
+} from '../../../common/resources';
+import VoucherEditForm from './VoucherEditForm';
+
+const VoucherEditContainer = ({
+  match: { params },
+  mutator,
+  resources,
+  history,
+}) => {
+  const [voucher, setVoucher] = useState();
+  const vendorInvoiceNo = get(resources, ['invoice', 'records', 0, 'vendorInvoiceNo']);
+  const { allowVoucherNumberEdit } = getConfigSetting(get(resources, 'configVoucherNumber.records', {}));
+  const showCallout = useShowCallout();
+
+  useEffect(() => {
+    mutator.voucher.GET({
+      params: {
+        id: params.voucherId,
+      },
+    })
+      .then(setVoucher)
+      .catch(() => {
+        showCallout({ messageId: 'ui-invoice.errors.voucherLoadingFailed', type: 'error' });
+      });
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const closeVoucherForm = useCallback(
+    () => {
+      const _path = `/invoice/view/${params.id}/voucher/${params.voucherId}/view`;
+
+      history.push(_path);
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [params.id],
+  );
+
+  const saveVoucher = useCallback(
+    (vouch) => {
+      mutator.voucher.PUT(vouch)
+        .then(() => showCallout({ messageId: 'ui-invoice.invoice.details.voucher.save.success' }))
+        .then(closeVoucherForm)

```

Comment:
if you don't use async things, maybe it makes sense to put this call into the previous `then`. But if you have good arguments I would be glad to discuss further